src/ng/compile.js

```diff
@@ -1877,6 +1877,36 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         return value || null;
       }
 
+      function setupControllers($element, attrs, transcludeFn, controllerDirectives, isolateScope, scope) {
```

Comment:
Actually, if `newIsolateScopeDirective` is not truthy then `isolateScope` will be null, in which case we don't need `newIsolateScopeDirective` to be passed through at all, via param or closure.